File: DesignPatternsExercises/java/FlyweightDesignPattern/Exercise2/FlightWorldMap.java
import java.util.Random;

public class FlightWorldMap {
    private final String[] flights = { "Air France", "American Airlines", "BA", "Delta", "KLM", "United Airlines" };
    private Random random = new Random();
    private FlightFactory flightFactory = new FlightFactory();

    public void displayAllFlights() {
        for (int i = 0; i < 10; i++) {
            Flight flight = flightFactory.__________(4)__________(getRandomFlight());
            flight.drawOnMap(getRandomFlightNumber(), getRandomLatitude(), getRandomLongitude());
        }
    }

    private double getRandomLatitude() {
        double value = Math.random() * 90;
        return random.nextBoolean() ? value : value * -1D;
    }

    private double getRandomLongitude() {
        double value = Math.random() * 180;
        return random.nextBoolean() ? value : value * -1D;
    }

    private String getRandomFlight() {
        return flights[random.nextInt(5)];
    }

    private int getRandomFlightNumber() {
        return random.nextInt(999);
    }
}

File: DesignPatternsExercises/ruby/flyweight_design_pattern/flyweight_design_pattern2/exercise/flight_world_map.rb
require_relative 'flight_factory'

class FlightWorldMap
  def initialize
    @flights = ['Air France', 'American Airlines', :BA, :Delta, :KLM, 'United Airlines']
    @factory = __________(7)__________
  end

  def __________(8)__________
    10.times do
      flight = __________(9)__________.flight(random_flight)
      flight.draw_on_map(random_flight_number, random_latitude, random_longitude)
    end
  end

  def random_latitude
    rand(0..90)
  end

  def random_longitude
    rand(0..180)
  end

  def random_flight
    @flights[rand(0..5)]
  end

  def __________(10)__________
    rand(100..900)
  end
end
